File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    let missing = [];
    
    for (let i = 0; i < nums.length; i++) {
        const num = Math.abs(nums[i]);
        const index = num-1;
        nums[index] = Math.abs(nums[index]) * -1;
    }
    
    
    for (let j = 0; j < nums.length; j++) {
        if (nums[j] > 0) missing.push(j+1);
    }
    
    return missing;
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        res = []
        numSet = set()
        for n in nums:
            numSet.add(n)
        for i in range(1, len(nums)+1):
            if i not in numSet:
                res.append(i)
        return res